napari/utils/translations.py

```diff
@@ -0,0 +1,429 @@
+"""
+Localization utilities to find available language packs and packages with
+localization data.
+"""
+
+import gettext
+import os
+from pathlib import Path
+
+# Entry points
+NAPARI_LANGUAGEPACK_ENTRY = "napari.languagepack"
+
+# Constants
+DEFAULT_LOCALE = "en"
+LOCALE_DIR = "locale"
+
+
+def _get_display_name(
+    locale: str, display_locale: str = DEFAULT_LOCALE
+) -> str:
+    """
+    Return the language name to use with a `display_locale` for a given language locale.
+
+    This is used to generate the preferences dialog options.
+
+    Parameters
+    ----------
+    locale: str
+        The language name to use.
+    display_locale: str, optional
+        The language to display the `locale`.
+
+    Returns
+    -------
+    str
+        Localized `locale` and capitalized language name using `display_locale` as language.
+    """
+    # This is a dependency of the language packs to keep out of core
+    import babel
+
+    locale = locale if _is_valid_locale(locale) else DEFAULT_LOCALE
+    display_locale = (
+        display_locale if _is_valid_locale(display_locale) else DEFAULT_LOCALE
+    )
+    loc = babel.Locale.parse(locale)
+    return loc.get_display_name(display_locale).capitalize()
+
+
+def _is_valid_locale(locale: str) -> bool:
+    """
+    Check if a `locale` value is valid.
+
+    Parameters
+    ----------
+    locale: str
+        Language locale code.
+
+    Notes
+    -----
+    A valid locale is in the form language (See ISO-639 standard) and an
+    optional territory (See ISO-3166 standard).
+
+    Examples of valid locales:
+    - English: DEFAULT_LOCALE
+    - Australian English: "en_AU"
+    - Portuguese: "pt"
+    - Brazilian Portuguese: "pt_BR"
+
+    Examples of invalid locales:
+    - Australian Spanish: "es_AU"
+    - Brazilian German: "de_BR"
+    """
+    # This is a dependency of the language packs to keep out of core
+    import babel
```

Comment:
how would this end up getting installed in typically usage?  Is this something that all language packs will necessarily have in their `install_requires`?  Do we need to add an extra?

Reply:
>Is this something that all language packs will necessarily have in their install_requires?

This was my suggestion, but the language packs template will be generated automatically on the `language-packs` repo, so we will take care of that anyway.